Parsers/ASimAuditEvent/Parsers/vimAuditEventAuth0.yaml.yaml

```diff
@@ -0,0 +1,187 @@
+     "gd_send_voice","Other","Informational","Success",
+     "gd_send_voice_failure","Other","Informational","Failure",
+     "gd_start_enroll","Other","Informational","Success",
+     "gd_start_enroll_failed","Other","Informational","Failure",
+     "gd_tenant_update","Other","Informational","Success",
+     "gd_unenroll","Other","Informational","Success",
+     "gd_update_device_account","Other","Informational","Success",
+     "gd_webauthn_challenge_failed","Other","Informational","Failure",
+     "gd_webauthn_enrollment_failed","Other","Informational","Failure",
+     "limit_delegation","Other","Informational","Success", 
+     "pla","Other","Informational","Success",
+     "resource_cleanup","Other","Informational","Success",
+     "sapi","Create","Informational","Success",
+     "sce","Other","Informational","Success",
+     "scp","Other","Informational","Success",
+     "scpn","Other","Informational","Success",
+     "scpr","Other","Informational","Success",
+     "scu","Other","Informational","Success",
+     "sd","Other","Informational","Success",
+     "sdu","Other","Informational","Success",
+     "si","Other","Informational","Success",
+     "signup_pwd_leak","Other","Informational","Failure",
+     "srrt","Other","Informational","Success",  
+     "ss","Other","Informational","Success",
+     "sui","Other","Informational","Success",
+     "sv","Other","Informational","Success",
+      "svr","Other","Informational","Success",
+     ];
+     let Auth0AuditEvent=Auth0AuditEventLookup
+     |project EventOriginalResultDetails
+     |summarize makeset(EventOriginalResultDetails);
+     Auth0AM_CL
+        |where not(disabled)
+        |where type_s in (Auth0AuditEvent)
+              and (isnull(starttime) or TimeGenerated >= starttime) 
+          and (isnull(endtime) or TimeGenerated <= endtime)          
+          and ((array_length(srcipaddr_has_any_prefix) == 0) or (has_any_ipv4_prefix(ip_s,srcipaddr_has_any_prefix)))
+        //username, Eventtype and eventresult done down in the parser
+     |project TimeGenerated,
+              EventCount=int(1),
+              EventStartTime=date_t,
+              EventEndTime=TimeGenerated,
+              EventProduct='Auth0',
+              EventVendor='Okta',
+              EventSchema='AuditEvent',
+              EventSchemaVersion='0.1',
+              EventUid=_ItemId,
+              DvcFQDN=hostname_s,         
+              DvcIpAddr=ip_s,
+              DvcDomainType='FQDN',
+              DvcId=client_id_s,         
+              DvcAction=scope_s,         
+              EventMessage=description_s,         
+              EventOriginalResultDetails=type_s,
+              EventProductVersion='4.3.1',
+              ActorUsername=user_name_s,
+              ActorUsernameType='UPN',
+              ActorScope=tolower(user_name_s),
+              ActorUserIdType='Oktaid',
+              EventOriginalUid=_id_s,
+              Object=client_name_s,
```

Comment:
this should have Object detail. e.g.
For type_s 'sapi' and description_s 'Update Brute-force settings' Object would be 'Brute Force Setting' ObjectType as 'Configuration Atom'

Reply:
Can incorporate, please confirm on the event description approach as mentioned in the erstwhile comments, do we want to make that enumerated or fetch it from the Auth0AM _CL table?

Reply:
we can evaluate EventMessage first using both datatable and description_s approach. And then set Object details based on EventMessage. This way Object won't be null when description_s is not available.